src/radical/saga/attributes.py

```diff
@@ -741,7 +741,7 @@ def _attributes_t_underscore (self, key, force=False) :
 
 
         if  force or d['camelcasing'] :
-            temp = Attributes._camel_case_regex_1.sub(r'\1_\2', key)
+            temp = Attributes._camel_case_regex_1.sub(r'\1_\2', str(key))
             return Attributes._camel_case_regex_2.sub(r'\1_\2', temp).lower()
```

Comment:
Where was this an error?  And why only on this call? Should we re-enable the type checks (see line 726)?  But they are costly...

Reply:
Are type casting costly in Python 3? If not then we can keep it to be sure that it is a string.

Reply:
It's not about cost - but if we do it this way, we will be chasing after this problem endlessly.  Now you provided a fix for this call - but the next user might use the dict key in another call.  Or in another class.  We will have to sprinkle `str()` all over the place.  I mean, we can do that, but then we'd rather do it coordinated like:
```
def some_method(key, size):
   key=str(key)
   size=int(size)
   ...
```
How do you handle cases where the type can be more than one?
```
def some_method(key, size, data=None):
    ...
    if not ininstance(data, type(None), int, basestring):
       data = str(data)
```
Then we can just as well use the type check decorators again.
It is not that your fix won't work - it will.  But it does not remove the cause of the problem, only the effect in a specific case..